### PR TITLE
fix: [PL-40459]: migrate delegate tokens to encrypt the token value

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
## Describe your changes
Summary - 
We will go though all the delegate Tokens entries and create a new encryptedRecord which encrypts the delegateValue. We will delete the existing encryptedRecord from the encryptedTokenId value of the document. We will now set the created newly encryptedRecords Id as the encryptedTokenId field for the delegate token entry. This will remove all the stale entires that would have been generated because of the wrong migration which ran earlier.

Testing:
Local- 
1. Verified on creation of new delegateToken, an entry with encryptedRecordId field is being created or not.
2. Ran the migration and verified whether the previous/existing delegateTokens encryptedRecordId is updated or not and also verified the deletion of existing encryptedRecords.
3. Added sufficient logs and try/catch blocks to figure out any kind of errors if were to happen in other env's.

- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51325)

### CODE REVIEW

**Comment by jennyjames21** [comment]
LGTM

**Review by jennyjames21 — DISMISSED** [review]

**Comment by VikasMaddukuriHarness** [comment]
trigger uts

File: 390-db-migration/src/main/java/io/harness/migrations/all/EncryptDelegateTokenAndStoreAsNewFieldAndDeleteExistingRecordsMigration.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          if (delegateToken.getOwner() != null) {
+            String orgId =
+                DelegateEntityOwnerHelper.extractOrgIdFromOwnerIdentifier(delegateToken.getOwner().getIdentifier());
+            String projectId =
+                DelegateEntityOwnerHelper.extractProjectIdFromOwnerIdentifier(delegateToken.getOwner().getIdentifier());
+            tokenIdentifier = String.format("%s_%s_%s", delegateToken.getName().replace("/", "_"), orgId, projectId);
+          }
+          String encryptedTokenId =
+              delegateSecretManager.encrypt(delegateToken.getAccountId(), delegateToken.getValue(), tokenIdentifier);
+          if (null != delegateToken.getEncryptedTokenId()) {
```
Comment (xingchi-jin) [inline 390-db-migration/src/main/java/io/harness/migrations/all/EncryptDelegateTokenAndStoreAsNewFieldAndDeleteExistingRecordsMigration.java:87]: can we move this if block after line 93. This way the update happens before deletion. Safer

**Review by xingchi-jin — DISMISSED** [review]

**Comment by VikasMaddukuriHarness** [comment]
trigger uts

**Review by xingchi-jin — APPROVED** [review]

**Pull request merged**